src/crdb/mpl.py

```diff
@@ -42,6 +43,58 @@ def draw_table(table, factor=1.0, label=None, sys_lw=5, **kwargs):
     return lines
 
 
+def get_mean_datetime(timerange):
+    """
+    Return the average time for a given time range.
+
+    Parameters
+    ----------
+    timerange : string
+        CRDB time range in "YYYY/MM/DD-HHMMSS:YYYY/MM/DD-HHMMSS" format.
+    """
+    s1, s2 = timerange.split(":")
+    dt1 = datetime.strptime(s1, "%Y/%m/%d-%H%M%S")
+    dt2 = datetime.strptime(s2, "%Y/%m/%d-%H%M%S")
+    return dt1 + (dt2 - dt1)/2
```

Comment:
Unfortunately, this function does not always work. There are entries in the DB which contain multiple ranges separated by colons. It probably always works if you apply it only to time-series data, but if the function is applied to other data, it can fail.

How should we handle this? I believe we should raise a ValueError, because the result is probably not meaningful if it is computed from multiple ranges. The other option is to silently compute the mean of the means of the subranges.

In the function `draw_timeseries`, I propose to reject points which have more than one time range and raise a warning if the input table contains such data.

Reply:
I implemented these changes, so if you agree with this solution, just say so and I will merge this PR. Thank you very much for this high-quality contribution with nice docstring.